streaming/base/dataset/streaming.py

```diff
@@ -0,0 +1,639 @@
+            if world.is_local_leader:
+                shm.unlink()
+            return epoch, []
+
+        # Load the correct epoch and previous training sessions this epoch.
+        epoch = obj['epoch']
+        old_sessions = [np.array(x) for x in obj['sessions']]
+        return epoch, old_sessions
+
+    def _create_cur_session(self, epoch: int) -> Tuple[NDArray[np.int64], SharedMemory]:
+        """Create the current session, as we have just started an epoch.
+
+        Called by __iter__() in a worker process, or a per-rank process if workers aren't used.
+
+        Note: this also returns the underlying shared memory object, because the returned array
+        will become invalidated when it goes out of scope.
+
+        Args:
+            epoch (int): The current epoch.
+
+        Returns:
+            Tuple[NDArray[np.int64], SharedMemory]: Session and handle to shared memory.
+        """
+        world = World()
+        shm_name = f'{self._prefix}_session_{epoch}'
+        shm_bytes = world.num_workers * np.int64().nbytes
+        try:
+            shm = SharedMemory(shm_name, True, shm_bytes)
+        except FileExistsError:
+            shm = SharedMemory(shm_name)
+        cur_session = np.ndarray(world.num_workers, buffer=shm.buf, dtype=np.int64)
+        cur_session[:] = 0
+        return cur_session, shm
+
+    def _lookup_cur_session(
+            self, epoch: int) -> Tuple[Optional[NDArray[np.int64]], Optional[SharedMemory]]:
+        """Look up the current session, which exists if we are currently training.
+
+        Called by state_dict() in a per-rank process.
+
+        Note: this also returns the underlying shared memory object (if it exists), because the
+        returned array will become invalidated when it goes out of scope.
+
+        Args:
+            epoch (int): The current epoch.
+
+        Returns:
+            Tuple[Optional[NDArray[np.int64]], Optional[SharedMemory]]: Maybe session, maybe shm.
+        """
+        shm_name = f'{self._prefix}_session_{epoch}'
+        try:
+            shm = SharedMemory(shm_name)
+        except:
+            return None, None
+        num_workers = shm.size // np.int64().nbytes
+        cur_session = np.ndarray(num_workers, buffer=shm.buf, dtype=np.int64)
+        return cur_session, shm
+
+    def _get_partition(self, epoch: int, sessions: List[NDArray[np.int64]],
+                       world: World) -> Optional[NDArray[np.int64]]:
+        # Local leader generates the partitions.
+        if world.is_local_leader:
+            sequences = get_epoch(self.shard_sizes, self.shuffle, self.seed, epoch, sessions)
+            base = world.node * world.ranks_per_node * world.workers_per_rank
+            for rank_of_node in range(world.ranks_per_node):
+                for worker_of_rank in range(world.workers_per_rank):
+                    worker = base + rank_of_node * world.workers_per_rank + worker_of_rank
+                    name = f'{self._prefix}_part_{worker:03}'
```

Comment:
We can move all the naming line to a `getter()` method ? This way it would be easy to understand and modify at later point since it would be at just one place, i.e. in a `getter()` method